### ISSUE TITLE
alluc

### ISSUE THREAD

**Issue by chataigne73** [LordVenom/venom-xbmc-addons-beta#99]
bonjour

serait il plus performant pour alluc d'utiliser l'api comme fait ici : 
https://github.com/tknorris/salts/blob/9fea386fd12ff4a1bcdbb2bfc7f9275ca5454d3a/scrapers/alluc_scraper.py

on obtient tous les résultats en une seule requête, on peut obtenir 100 résultats d'un coup ...

qu'en pensez vous ?

**Comment by TmpName** [comment]
Yep, tu as raisons, m'en occuperais plus tard si personne ne s'y met.

**Comment by chataigne73** [comment]
Je m'y mets car je trouve ça intéressant

**Comment by TmpName** [comment]
Ok, je te laisse la place alors ^^.

**Comment by chataigne73** [comment]
à y regarder de plus prêt, je trouve que ça n'apporte rien fonctionellement pour l'utilisateur final
donc pas sûr que ça vaille le coup

**Comment by TmpName** [comment]
Non pour lui, ca va pas changer grand chose, sinon j'ai pas encore regardé, mais 
1 - le site peut changer, pas leur api
2 - le code sera plus propre, plus facile a faire.
3 - plus rapide.

Mais c'est clair que c'est pas une urgence, on peut attendre la prochaine modif de leur site pour avoir une bonne raison de le faire.

**Comment by LordVenom** [comment]
Si l'envie de "jouer" avec des api t'interesse regarde du coter de bêtaserie ou track voir si ont peux travailler avec un des deux, je préfére betaserie mais bon beaucoup utilise track.

**Comment by LordVenom** [comment]
J'ai modifier filmsvostfr pour inclure une nouvelle option, juste un "Plus" qui effectue une recherche des hosts sur alluc. Vous en pensez quoi ?

**Comment by TmpName** [comment]
Bonne question, y a un truc a faire la.

1 - Deja ca marche nickel
2 - C'est tres, tres rapide pour avoir d'autres hosts
3 - Plus visible et pratique que le menu contextuel (apres je sais pas si c'est mieux)
4 - discret (meme si en menu contextuel c'est encore plus discret, trop meme)

Par contre
1 - je passerais plutot par le module de recherche de alluc, plutot qu'une url car le site peut changer et alluc est capable de gerer les noms de series EXXSXX dans les recherche, d'ailleurs vais surement ameliorer le fichier site pour ca d'ailleurs.
2 - pourquoi ne pas proposer egalement la recherche globale via le moteur de vstream (ca remplacerait la "recherche similaire", on peut meme recuperer le dialog de la recherche visuelle.

En resumé, pour moi ca va me servir, ca me serait meme indispensable, quand je zappe dans les sources, trouve une serie interessante, avec un episode manquant. Plus besoin de repartir en arriere dans la recherche, pour tout retaper au clavier.

**Comment by LordVenom** [comment]
Mieux que le context car si aucune source pour le film il y a quand même le "Plus" et une chance que alluc en trouve une.

le module de recherche ? j'ai pas compris la.
Oui on peux aussi lancer une recherche direct comme ça mais pas dans les sources puisque la recherche vstream ne donne pas les sources.

Oui j'ai le même problême quand il manque une source, c'est pk je cherche toujour manuellement pour avoir un lien dans historique. A avoir pour ameliorer l'historique.

**Comment by TmpName** [comment]
Ha oui, on aura pas les sources direct, j'avais pas vu ca comme ca. En fait ca revient a remettre en place la recherche similaire.

Par module de recherche je voulais dire le dialog qui apparait quand on fait une recherche sur la recherche visuelle, celui qui propose vstream ou alluc.

Tu penses a quoi pour l'historique ? Y rajouter la meme option ?

**Comment by LordVenom** [comment]
rien de spécial, pour le moment on vas partir sur le plus, plus leger et asser rapide.

**Comment by TmpName** [comment]
Au fait vu le retard de la MAJ de janvier, celle de fevrier aura lieu vers ou a peu pres, pour que je me bouge le cul pour ameliorer alluc avant pour gerer les series.

**Comment by LordVenom** [comment]
Si tu veut que je lance une pas de soucis mais pour le moment c'est pas prévus ça depend des changements.

**Comment by TmpName** [comment]
Ok, bon ben apres verification, y a meme pas besoin, ca marche deja, tu chosis une serie, un episode, affiche les hoster, clic sur "plus", et tu as deja les sources pour le meme episode.

Et je viens de voir autre chose, c'est peut etre meme plus simple de passer par leur site que par leur api, nottament a cause des images, elle ne sont pas dans la recherche de base, il faut refaire une requete pour chaque resultat.

**Comment by TmpName** [comment]
Je viens de corriger easyvid.
Tu penses que cela ferait trop lourd de les mettres en MAJ auto comme les fichiers sites ?
Un blocage par date ?
Et de re-rendre la MAJ possible par le clic sur l'icone, sans passer par le menu ?

**Comment by LordVenom** [comment]
Je regarde ça

**Comment by LordVenom** [comment]
Bon j'ai commencer a modifier , on fait quoi alors les sites et les sources en auto ?

**Comment by TmpName** [comment]
Sources avec le meme fonctionnement que les sites, a savoir l'icone qui apparait (max tout les 2 jours de memoire) et mise a jour en cliquant dessus sans avoir a passer par le menu de config.

Apres j'ai pas encore vu tes modifs, donc je sais pas encore quoi proposer pour eviter de plomber le serveur. Peut etre rajouter un message expliqaunt que si l'utilisateur ne rencontre pas de problemes, la maj est inutile ? Ou si le fichier a plus de 1 mois, pas de calcul md5, car fichier non mit a jour ?

**Comment by LordVenom** [comment]
j'avance mais c'est la que je m’aperçois du chargement que ça vas faire un test sur 91 fichier.

Si nous ont fait pas de mise a jour l'utilisateur n'aurais pas le lien, mais si ont fait une mise a jour sur 1 fichiers le code vas faire le test sur les 91 fichier pour trouver celui qui a etait modifier.

**Comment by TmpName** [comment]
J'y ai pensé 91 fichiers multiplie par le nombre de persones pouvant lancer ca, ca fait pas mal.

En utilisant les api github, ce serait pas plus leger ? Je connais pas trop les limitations
eg https://api.github.com/repos/LordVenom/venom-xbmc-addons-beta/contents/plugin.video.vstream/resources/sites
Tu as plus d'infos que necessaire en 1 seule requete.

**Comment by LordVenom** [comment]
Trop limiter leur api ont avais eu des probleme avec ça.

**Comment by TmpName** [comment]
Lol.

Ha oui c'est marrant ca correspond a rien ? Je continus a chercher.

**Comment by TmpName** [comment]
Bon ca me saoule aussi, en theorie, il suffirait de charger un fichier et au lieu de calculer son md5 le convertir avec la base64 pour comparer les tailles, en pratique j'arrive a avoir un resulat different a chauque essais.

Une autre idée, a la place du fichier updates.xml.md5, mettre dedans une chaine avec la liste des fichiers modifiés depuis la derniere mise a jour, suivi d'un chiffre, comme kepliz_comfilmvostrfropenloadyoutch4.

Ca peut servir de cle pour savoir si il y a eu des fichiers de chnagés, vu que tu peut incrementer le dernier chiffre.
Et lors des mise a jour, tu ne compare les fichiers que si celui ci se trouve dans la chaine. Ca devrait meme etre encore plus legé que la methode actuelle, en ne changeant que 4/5 lignes du code actuel.

**Comment by LordVenom** [comment]
Ok exit le os.path.getsize(filename) ou encore le os.stat(path) un simple mais compliquer file=open(filepath).read()
return len(file)

fonctionne.

Pour le hash il ont tout un system pour le calcul, Il ajoute des espaces, le nombre de lettre et ensuite le contenue bref inutilisable mais la j'ai la bonne taille oufff.

**Comment by TmpName** [comment]
Ha bon ? Put1 moi j'ai essayé le meme code sans succés, je sais pas ou me suis raté.

**Comment by LordVenom** [comment]
Je suis pas loin de la vérité la, donc on a deux scan online toute les 48h plus 2 scan lors de la mise a jour.

**Comment by TmpName** [comment]
J'attend la version finale pr tester, donc j'ai juste jeté un oeuil sur le code.
J'ai vu que tu as laissé des bouts du code avec le faux md5, tu as prevu de le virer ou de le laisser pour eviter le scan de fichiers toute les 48h ?

Faudra mettre sur un post l'url qui genere le fichier json, car moi je vais jamais m'en souvenir, ou en com dans le fichier python.

Tu as testé en supprimant la ligne 330 ?

Je pense qu'on peux meme passer a 3 jours au lieu de 2 pour les tests. Le fichier fonctionne le dimanche, deconne le lundi, le temps d'etre prevenu, de corriger, de tester, de remettre en ligne la nouvelle version, MAJ le mercredi je trouve pas ca exagere.

**Comment by LordVenom** [comment]
Voila le code est nettoyer, Plus de md5 tout ce fait par taille et pas les fichiers Json.

**Comment by TmpName** [comment]
Ca a l'air de marcher nickel, mais je peux pas tester sur linux par contre.
J'attend une mise a jour en reel pour te confirmer ca.

**Comment by Kodigoal** [comment]
Et avec cette nouvelle méthode, si vous rajoutez un nouveau site ou un nouveau host, il le rajoutera de notre côté ?
Je crois qu'avec la méthode d'avant , il comparait que les fichiers présents de notre côté , sans rajouter
si yen a 1 de nouveau du côté github vstream.

**Comment by TmpName** [comment]
Nope, toujours pas, les rajouts ne se font que a la mise a jour, mais c'est normal vu qu'il faut aussi les rajouter au coeur de Vstream pour la recherche globale.
Si on rajoute juste le fichier, il n'apparaitra ni dans les options, ni dans la recherche.

**Comment by Kodigoal** [comment]
ah oui pas penser à ca :) mais super la, rapide ! merci

**Comment by TmpName** [comment]
Par contre, a ta place je ne ferais pas cette mise a jour car FlashX et netu ne sont pas encore sur la version off, donc si tu fais une mise a jour, tu va perdre les versions de la beta.

Je te dis ca car je viens de faire la connerie ^^.

**Comment by Kodigoal** [comment]
Mais on utilise  le lien de ' download_url ' dans les fichiers json non ?
il pointe  vers la beta je crois.
J'ai pas fait gaffe sur le coup.
J'ai remis ex version ou j'avais changé un peu le code, en mettant directement
un lien dans le home de vstream ,1 click et verif direct.
Faut que je reteste le nouveau :)
Toute facon par netu , mon ip est 'banni' , je sais pas pourquoi...
si je vais sur hqq. tv  error 16 access denied :)

**Comment by Kodigoal** [comment]
J'ai retesté pour bien comprendre et il cherche bien sur la beta les fichiers Maj.

Test 1) install vstream zip beta 10/02- verif Maj - il met a jour les 2 fichiers update le 3/02 ( netu/regarder_film) vu que les fichiers json date du 2/02, mes 2 fichiers sont plus récents en local donc
il les retelecharge (même si c'était les mêmes).Normal.

test 2) copie/colle dossier sites/hosters d'un vieux zip du 21 janvier.
            il Maj tous les Fichiers update jusqu'au 2/02 , pas ceux du 3/02 ( ont pas
            changé depuis le 21/02).Normal.

test 3) création des sites&hosts .json au 10/02 , heberger sur un site, changement url dans le script,
            il Maj bien les 2 fichiers update du 3/02.( sans le dernier site ajouté , qui n'est pas présent en local).Normal.
donc si  j'ai bien tout compris, ca le fait.
Niquel ce système :)

**Comment by TmpName** [comment]
Ha oui, j'avais pas vu, je sais pas si c'est fait expres ou pas, vu que pour le moment c'est encore en test.
Actuellement il met a jour en fonction des fichiers sur le beta, alors que avant c'etait ceux de la version officielle (mais hors repositary), a voir si c'est en test ou en definitif avec LordVenom.

**Comment by LordVenom** [comment]
Pour des test uniquement . je met la bonne URL bientôt.

**Comment by TmpName** [comment]
Quelqu'un a pu tester sur un systeme sous linux ou pire android ?

Et d'apres le code il y a meme la possibilitée de mettre uniquement les fichier modifiés dans le fichier json, ca demande plus de taff a celui qui fais la mise a jour, mais c'est egalement possible avec le code actuel.

**Comment by Kodigoal** [comment]
j'avais testé sous android, ca fonctionne.
je peux tester sous linux si besoin.
Et si vous mettez l'url par l'api directement dans le code , ca evite le copier/coller
lors des update, ca serait pas possible?
Après, ya pêtre des limites niveau api..

**Comment by Kodigoal** [comment]
j'avais testé sous android, ca fonctionne.
je peux tester sous linux si besoin.
Et si vous mettez l'url par l'api directement dans le code , ca evite le copier/coller
lors des update, ca serait pas possible?
Après, ya pêtre des limites niveau api..
edit: Testé sous linux /kodi 14.2 , ca fonctionne.

**Comment by Kodigoal** [comment]
j'avais testé sous android, ca fonctionne.
je peux tester sous linux si besoin.
Et si vous mettez l'url par l'api directement dans le code , ca evite le copier/coller
lors des update, ca serait pas possible?
Après, ya pêtre des limites niveau api..
edit: Testé sous linux (Fedora 21 64bits/kodi 14.2 ),  ca fonctionne.

**Comment by TmpName** [comment]
Et tu peux confirmer que ca ne telecharge que les nouveaux fichiers, pas la totalité ?

Oui, je pense que c'est pour eviter de charger github, tu imagines si tout les users font un appel a l'api a chaque fois, surtout que si ca se trouve leur serveur recalcule les md5 a chaque fois.

Apres il n'y a pas 1 maj tout les jours, la ca fait 1 bonne semaine que j'ai rien touche, moi le copier/coller me derange pas et en cas de bugs, il y a la possibilité de tout bloquer en mettant un fichier vide.

**Comment by Kodigoal** [comment]
oui ca telecharge bien que les nouveaux fichiers.
ok pour l'api! :)

**Comment by LordVenom** [comment]
A verifier.
